src/rust/ensogl/lib/text/src/component/area.rs

```diff
@@ -692,6 +692,15 @@ impl Area {
         //TODO[ao] we cannot move selection symbol, as it is global for all the text areas.
         SmallVec::from_buf([text_symbol,/*selection_symbol*/])
     }
+
+    /// Set the text color based on the given theme path. Updates the existing text and future text.
+    pub fn set_text_color_from_style(&self, path:&display::style::path::Path) {
+        let style      = StyleWatch::new(&self.data.app.display.scene().style_sheet);
+        let text_color = style.get_color(path);
+        let text_color = color::Rgba::from(text_color);
+        self.frp.set_color_all.emit(text_color);
+        self.frp.set_default_color.emit(text_color);
+    }
```

Comment:
I would remove it and I would compute the color on call side instead. The reason is that we are in the middle of the transition to FrpStyleWatch, which will be integrated with Frp on the caller side. This function is designed explicitly in such way, that it has to create and destroy instance of StyleWatch, so this function would need to be deleted when we will be finishing the transition. I'd prefer to not create functions that we already know are in opposite to the new design - just to ake the transition / refactoring easier.